### PR TITLE
added trio options

### PR DESCRIPTION
**Pull request by yumisims**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by GallVp** [comment]
Thank you for the contribution. Might be related to https://github.com/nf-core/modules/pull/5632

File: modules/nf-core/merquryfk/merquryfk/main
Comment (SPPearce) [inline modules/nf-core/merquryfk/merquryfk/main:1]: This file shouldn't be here?

File: modules/nf-core/merquryfk/merquryfk/main.nf [hunk 1]
```diff
@@ -37,14 +45,18 @@ process MERQURYFK_MERQURYFK {
     }
     def args = task.ext.args ?: ''
     prefix = task.ext.prefix ?: "${meta.id}"
+    def mat_ktab = matktab ? "${matktab.find{ it.toString().endsWith(".ktab") }}" : ''
```
Comment (SPPearce) [inline modules/nf-core/merquryfk/merquryfk/main.nf:48]: Does this need the `find`? Is it not just one file?

File: modules/nf-core/merquryfk/merquryfk/main.nf [hunk 2]
```diff
@@ -37,14 +45,18 @@ process MERQURYFK_MERQURYFK {
...
+    def pat_ktab = patktab ? "${patktab.find{ it.toString().endsWith(".ktab") }}" : ''
     def FASTK_VERSION = 'f18a4e6d2207539f7b84461daebc54530a9559b0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
     def MERQURY_VERSION = '8ae344092df5dcaf83cfb7f90f662597a9b1fc61' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
     """
     MerquryFK \\
         $args \\
         -T$task.cpus \\
         ${fastk_ktab.find{ it.toString().endsWith(".ktab") }} \\
-        $assembly \\
+        ${mat_ktab} \\
```
Comment (SPPearce) [inline modules/nf-core/merquryfk/merquryfk/main.nf:57]: Does the tool not need some kind of argument to know what file is what?

File: modules/nf-core/merquryfk/merquryfk/main.nf [hunk 3]
```diff
@@ -55,4 +67,20 @@ process MERQURYFK_MERQURYFK {
...
     END_VERSIONS
     """
+    stub:
+    prefix = task.ext.prefix ?: "${meta.id}"
+    def FASTK_VERSION = 'f18a4e6d2207539f7b84461daebc54530a9559b0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def MERQURY_VERSION = '8ae344092df5dcaf83cfb7f90f662597a9b1fc61' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    """
+    touch ${prefix}.completeness.stats
+    touch ${prefix}.qv
+    touch ${prefix}._.qv
```
Comment (SPPearce) [inline modules/nf-core/merquryfk/merquryfk/main.nf:77]: Why is this name `._.`?

**Review by SPPearce — COMMENTED** [review]
Having added nf-test, you can remove the pytest tests (everything under `tests/modules/nf-core/merquryfk/merquryfk`), as well as the lines from `tests/pytest_config.yml`

**Comment by SPPearce** [comment]
Are the image outputs either pdf or png depending on an argument? If so, can you just make them one channel that holds either.

File: modules/nf-core/merquryfk/merquryfk/main.nf (hunk 1 above)
Comment (yumisims) [reply modules/nf-core/merquryfk/merquryfk/main.nf:48]: yes, merquryfk -t generates a collection of files.

File: modules/nf-core/merquryfk/merquryfk/main.nf (hunk 2 above)
Comment (yumisims) [reply modules/nf-core/merquryfk/merquryfk/main.nf:57]: the usuage suggests the two optional ktabs comes after the read ktab.

File: modules/nf-core/merquryfk/merquryfk/main.nf (hunk 1 above)
Comment (SPPearce) [reply modules/nf-core/merquryfk/merquryfk/main.nf:48]: Right, that is weird, but I see that is what the fastk tool is doing.

File: modules/nf-core/merquryfk/merquryfk/main.nf (hunk 3 above)
Comment (yumisims) [reply modules/nf-core/merquryfk/merquryfk/main.nf:71]: assembly qv file is another qv file that needs to be in the output, its pattern is usually ${prefix}.*.qv, what do you suggest to change to please? @SPPearce . thanks

Comment (SPPearce) [reply modules/nf-core/merquryfk/merquryfk/main.nf:71]: Ok, if that is the sort of name that gets generated out of the tool that is fine.

File: modules/nf-core/merquryfk/merquryfk/main [hunk 4]
```diff
@@ -0,0 +1,82 @@
+process MERQURYFK_MERQURYFK {
```
Comment (mahesh-panchal) [inline modules/nf-core/merquryfk/merquryfk/main:1]: This extra file needs to be removed.

File: modules/nf-core/merquryfk/merquryfk/main.nf [hunk 5]
```diff
@@ -6,25 +6,27 @@ process MERQURYFK_MERQURYFK {
     container 'ghcr.io/nbisweden/fastk_genescopefk_merquryfk:1.2'
 
     input:
-    tuple val(meta), path(fastk_hist), path(fastk_ktab), path(assembly), path(haplotigs)
+    tuple val(meta), path(fastk_hist),path(fastk_ktab),path(asm1),path(haplotigs)
```
Comment (mahesh-panchal) [inline modules/nf-core/merquryfk/merquryfk/main.nf:9]: I would prefer the spacing here be kept as it was

File: modules/nf-core/merquryfk/merquryfk/main.nf [hunk 6]
```diff
@@ -6,25 +6,27 @@ process MERQURYFK_MERQURYFK {
...
     tuple val(meta), path("${prefix}.*.qv")               , emit: assembly_qv
-    tuple val(meta), path("${prefix}.*.spectra-cn.fl.png"), emit: spectra_cn_fl_png,  optional: true
-    tuple val(meta), path("${prefix}.*.spectra-cn.fl.pdf"), emit: spectra_cn_fl_pdf,  optional: true
-    tuple val(meta), path("${prefix}.*.spectra-cn.ln.png"), emit: spectra_cn_ln_png,  optional: true
-    tuple val(meta), path("${prefix}.*.spectra-cn.ln.pdf"), emit: spectra_cn_ln_pdf,  optional: true
-    tuple val(meta), path("${prefix}.*.spectra-cn.st.png"), emit: spectra_cn_st_png,  optional: true
-    tuple val(meta), path("${prefix}.*.spectra-cn.st.pdf"), emit: spectra_cn_st_pdf,  optional: true
+    tuple val(meta), path("${prefix}.*.spectra-cn.fl.*")  , emit: spectra_cn_fl,  optional: true
+    tuple val(meta), path("${prefix}.*.spectra-cn.ln.*")  , emit: spectra_cn_ln,  optional: true
+    tuple val(meta), path("${prefix}.*.spectra-cn.st.*")  , emit: spectra_cn_st,  optional: true
```
Comment (mahesh-panchal) [inline modules/nf-core/merquryfk/merquryfk/main.nf:19]:
```suggestion
    tuple val(meta), path("${prefix}.*.spectra-cn.fl.{png,pdf}")  , emit: spectra_cn_fl,  optional: true
    tuple val(meta), path("${prefix}.*.spectra-cn.ln.{png,pdf}")  , emit: spectra_cn_ln,  optional: true
    tuple val(meta), path("${prefix}.*.spectra-cn.st.{png,pdf}")  , emit: spectra_cn_st,  optional: true
```

File: modules/nf-core/merquryfk/merquryfk/main.nf [hunk 7]
```diff
@@ -6,25 +6,27 @@ process MERQURYFK_MERQURYFK {
...
     tuple val(meta), path("${prefix}.qv")                 , emit: qv
-    tuple val(meta), path("${prefix}.spectra-asm.fl.png") , emit: spectra_asm_fl_png, optional: true
-    tuple val(meta), path("${prefix}.spectra-asm.fl.pdf") , emit: spectra_asm_fl_pdf, optional: true
-    tuple val(meta), path("${prefix}.spectra-asm.ln.png") , emit: spectra_asm_ln_png, optional: true
-    tuple val(meta), path("${prefix}.spectra-asm.ln.pdf") , emit: spectra_asm_ln_pdf, optional: true
-    tuple val(meta), path("${prefix}.spectra-asm.st.png") , emit: spectra_asm_st_png, optional: true
-    tuple val(meta), path("${prefix}.spectra-asm.st.pdf") , emit: spectra_asm_st_pdf, optional: true
+    tuple val(meta), path("${prefix}.spectra-asm.fl.*")   , emit: spectra_asm_fl, optional: true
+    tuple val(meta), path("${prefix}.spectra-asm.ln.*")   , emit: spectra_asm_ln, optional: true
+    tuple val(meta), path("${prefix}.spectra-asm.st.*")   , emit: spectra_asm_st, optional: true
```
Comment (mahesh-panchal) [inline modules/nf-core/merquryfk/merquryfk/main.nf:23]:
```suggestion
    tuple val(meta), path("${prefix}.spectra-asm.fl.{png,pdf}")   , emit: spectra_asm_fl, optional: true
    tuple val(meta), path("${prefix}.spectra-asm.ln.{png,pdf}")   , emit: spectra_asm_ln, optional: true
    tuple val(meta), path("${prefix}.spectra-asm.st.{png,pdf}")   , emit: spectra_asm_st, optional: true
```

File: modules/nf-core/merquryfk/merquryfk/main.nf [hunk 8]
```diff
@@ -6,25 +6,27 @@ process MERQURYFK_MERQURYFK {
     container 'ghcr.io/nbisweden/fastk_genescopefk_merquryfk:1.2'
 
     input:
-    tuple val(meta), path(fastk_hist), path(fastk_ktab), path(assembly), path(haplotigs)
+    tuple val(meta), path(fastk_hist),path(fastk_ktab),path(asm1),path(haplotigs)
+    path matktab                                                                        //optional
+    path patktab                                                                        //optional
```
Comment (mahesh-panchal) [inline modules/nf-core/merquryfk/merquryfk/main.nf:11]: Usage wise, would these normally be a paired tuple? Or does it make it problematic having two lists of files as a tuple?

File: modules/nf-core/merquryfk/merquryfk/main.nf [hunk 9]
```diff
@@ -37,14 +39,18 @@ process MERQURYFK_MERQURYFK {
...
     def MERQURY_VERSION = '8ae344092df5dcaf83cfb7f90f662597a9b1fc61' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
     """
     MerquryFK \\
         $args \\
         -T$task.cpus \\
         ${fastk_ktab.find{ it.toString().endsWith(".ktab") }} \\
-        $assembly \\
+        ${mat_ktab} \\
+        ${pat_ktab} \\
+        $asm1 \\
```
Comment (mahesh-panchal) [inline modules/nf-core/merquryfk/merquryfk/main.nf:53]: Why was this changed to `asm1`? There's no other assembly to reference here.

File: modules/nf-core/merquryfk/merquryfk/meta.yml [hunk 10]
```diff
@@ -50,54 +58,30 @@ output:
...
-      description: "Unstacked assembly spectra filled plot in PNG format"
-      pattern: "*.spectra-asm.fl.png"
-  - spectra_asm_ln_png:
+  - spectra_cn_fl:
       type: file
-      description: "Unstacked assembly spectra line plot in PNG format"
-      pattern: "*.spectra-asm.ln.png"
-  - spectra_asm_st_png:
+      description: "Unstacked copy number spectra filled plot in PNG or PDF format"
+      pattern: "*.spectra-cn.fl.*"
```
Comment (mahesh-panchal) [inline modules/nf-core/merquryfk/merquryfk/meta.yml:64]:
```suggestion
      pattern: "*.spectra-cn.fl.{png,pdf}"
```
This should also be applied to the other patterns too.

File: modules/nf-core/merquryfk/merquryfk/main.nf [hunk 11]
```diff
@@ -6,25 +6,27 @@ process MERQURYFK_MERQURYFK {
...
-    tuple val(meta), path("${prefix}.spectra-asm.st.pdf") , emit: spectra_asm_st_pdf, optional: true
+    tuple val(meta), path("${prefix}.spectra-asm.fl.*")   , emit: spectra_asm_fl, optional: true
+    tuple val(meta), path("${prefix}.spectra-asm.ln.*")   , emit: spectra_asm_ln, optional: true
+    tuple val(meta), path("${prefix}.spectra-asm.st.*")   , emit: spectra_asm_st, optional: true
+    tuple val(meta), path("${prefix}.phased_block.bed")   , emit: phased_block_bed,   optional: true
+    tuple val(meta), path("${prefix}.phased_block.stats") , emit: phased_block_stats, optional: true
+    tuple val(meta), path("${prefix}.continuity.N.*")     , emit: continuity_N,       optional: true
+    tuple val(meta), path("${prefix}.block.N.*")          , emit: block_N,            optional: true
+    tuple val(meta), path("${prefix}.block.blob.*")       , emit: block_blob,         optional: true
+    tuple val(meta), path("${prefix}.hapmers.blob.*")     , emit: hapmers_blob,       optional: true
```
Comment (mahesh-panchal) [inline modules/nf-core/merquryfk/merquryfk/main.nf:29]:
```suggestion
    tuple val(meta), path("${prefix}.continuity.N.{pdf,png}")     , emit: continuity_N,       optional: true
    tuple val(meta), path("${prefix}.block.N.{pdf,png}")          , emit: block_N,            optional: true
    tuple val(meta), path("${prefix}.block.blob.{pdf,png}")       , emit: block_blob,         optional: true
    tuple val(meta), path("${prefix}.hapmers.blob.{pdf,png}")     , emit: hapmers_blob,       optional: true
```

File: modules/nf-core/merquryfk/merquryfk/meta.yml [hunk 12]
```diff
@@ -106,7 +90,33 @@ output:
...
+      type: file
+      description: "Stacked assembly block plot in PNG or PDF format"
+      pattern: "*.block.blob.*"
+  - hapmers_blob:
+      type: file
+      description: "Stacked assembly hapmers block plot in PNG or PDF format"
+      pattern: "*.hapmers.blob.*"
 authors:
   - "@mahesh-panchal"
+  - "trio option added by @yumisims"
```
Comment (mahesh-panchal) [inline modules/nf-core/merquryfk/merquryfk/meta.yml:119]:
```suggestion
  - "@yumisims"
```
I think the extra text would just cause problems for anything automated that's using this.

**Review by mahesh-panchal — CHANGES_REQUESTED** [review]
I think it would be more useful to embed the options params in the tests rather than having several config files for nf-test.

File: modules/nf-core/merquryfk/merquryfk/main (hunk 4 above)
Comment (yumisims) [reply modules/nf-core/merquryfk/merquryfk/main:1]: it has been removed

File: modules/nf-core/merquryfk/merquryfk/main.nf (hunk 9 above)
Comment (yumisims) [reply modules/nf-core/merquryfk/merquryfk/main.nf:53]: that might have been a mistake, I could change it back.

File: modules/nf-core/merquryfk/merquryfk/main.nf (hunk 8 above)
Comment (yumisims) [reply modules/nf-core/merquryfk/merquryfk/main.nf:11]: I tried list, but failed at testing.

**Review by mahesh-panchal — COMMENTED** [review]
It seems using multiple configs is still the standard way, so ignore my previous comment about it. 
The only thing I'd like updated now is for the patterns to match in the meta.yml. 
I'm not quite sure what the issue is the environment.yml. I'll check up on that.

And thank you for making the update and converting to nf-test!

**Review by mahesh-panchal — APPROVED** [review]
Delete the environment.yml in the merquryfk/merquryfk directory and you should be good to go.
